### ISSUE TITLE
Replace renderer with Vulkan

### ISSUE THREAD

**Issue by mmmhj2** [graphics-study-group/needrename#5]
Replace render with Vulkan:

- [x] Reconfigure Cmake dependencies
- [x] Rewrite renderer bootstrapping
- [x] Redesign rendering pipeline (Command buffers, etc.)
  - [x] Redesign material and shader interface
  - [x] Enable shader compilation with Cmake
- [x] Rewrite texture and mesh interface (after #2)
- [x] Redesign rendering loop and related interface
  - [x] Render pass interfaces (or vk 1.3 dynamic rendering support)
  - [x] Render target interfaces (Framebuffers and ImageViews)
  - [x] Command buffer interfaces
  - [x] Better fixed function pipeline management
  - [x] Integration with pre-existing components (chiefly cameras)
- [x] (Postponed) Final code style audit

Possible further improvement:
1. Better render pass and dependencies handling
2. Better material system
3. Integration with asset system (textures, etc.)
4. Complete render-time constant handling (skeletal transforms, enviromental info, etc.)
